tests/MongoDB.Driver.Core.Tests/Core/Servers/ServerTests.cs

```diff
@@ -266,6 +267,145 @@ public void Dispose_should_dispose_the_server()
             channel.Should().NotBeNull();
         }
 
+        [Theory]
+        [ParameterAttributeData]
+        public void GetChannel_should_update_topology_and_clear_connection_pool_on_network_error_or_timeout(
+            [Values("timeout", "network")] string errorType,
+            [Values(false, true)] bool async)
+        {
+            var serverId = new ServerId(_clusterId, _endPoint);
+            var connectionId = new ConnectionId(serverId);
+            Exception innerMostException;
+            switch (errorType)
+            {
+                case "timeout":
+                    innerMostException = new SocketException((int)SocketError.TimedOut);
+                    break;
+                case "network":
+                    innerMostException = new SocketException((int)SocketError.NetworkUnreachable);
+                    break;
+                default: throw new ArgumentException("Unknown error type.");
+            }
+
+            var openConnectionException = new MongoConnectionException(connectionId, "Oops", new IOException("Cry", innerMostException));
+            var mockConnection = new Mock<IConnectionHandle>();
```

Comment:
Can some of these lines be removed taking into account that a similar initialization happens in the constructor of this test class?

Reply:
Unfortunately whenever these methods use a local instead of the version created in the instructor and stored as a private class variable,  it's because the mock requires different behavior :(